modules/custom/az_mail/az_mail.install

```diff
@@ -0,0 +1,34 @@
+<?php
+
+/**
+ * @file
+ * Install, update and uninstall functions for az_mail module.
+ */
+
+/**
+ * Implements hook_install().
+ */
+function az_mail_install() {
+  $config_factory = \Drupal::service('config.factory');
+  $mail_config = $config_factory->getEditable('system.mail');
+  $mail_system = $mail_config->get('interface.default');
+  if ($mail_system != 'SMTPMailSystem') {
+    $config = $config_factory->getEditable('smtp.settings');
+    $config->set('prev_mail_system', $mail_system);
```

Comment:
This probably wasn't working since it doesn't call `save()`.  My bad 🤦🏻‍♂️ .
```suggestion
    $config->set('prev_mail_system', $mail_system)->save();
```